### PR TITLE
PP-5463 Emit refund availability updated on capture

### PR DESCRIPTION
**Pull request by mrlumbu**
Refund availability changes either on capture or capture submitted, so
emit a refund availability updated event when either of these events occur

I have consolidated the event creation into a single event factory, which
means StateTransitionEmitterProcess no longer has to know about the
details of event creation at all, which seems right.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/events/model/EventFactory.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    private List<Event> createRefundEvents(RefundStateTransition refundStateTransition) throws EventCreationException {
+        RefundHistory refundHistory = refundDao.getRefundHistoryByRefundExternalIdAndRefundStatus(
+                refundStateTransition.getRefundExternalId(),
+                refundStateTransition.getRefundStatus())
+                .orElseThrow(() -> new EventCreationException(refundStateTransition.getIdentifier()));
+
+        Event refundEvent = createRefundEvent(refundHistory, refundStateTransition.getStateTransitionEventClass());
+        Optional<Event> refundAvailabilityEvent = createRefundAvailabilityUpdatedEvent(
+                refundHistory.getChargeEntity().getExternalId(),
+                refundHistory.getHistoryEndDate(),
```
Comment (jankowiakmaria) [inline src/main/java/uk/gov/pay/connector/events/model/EventFactory.java]: You've said that `historyEndDate` can be null. Why the change from `historyStartDate`?

File: src/test/java/uk/gov/pay/connector/events/model/refund/EventFactoryTest.java [hunk 2]
```diff
@@ -192,4 +209,94 @@ public void shouldCreateCorrectEventsFromRefundErrorStateTransition() throws Exc
...
+        List<Event> events = eventFactory.createEvents(paymentStateTransition);
+
+        assertThat(events.size(), is(2));
+        CaptureSubmitted event1 = (CaptureSubmitted) events.get(0);
+        Assert.assertThat(event1, instanceOf(CaptureSubmitted.class));
+
+        RefundAvailabilityUpdated event2 = (RefundAvailabilityUpdated) events.get(1);
+        Assert.assertThat(event2, instanceOf(RefundAvailabilityUpdated.class));
+        Assert.assertThat(event2.getEventDetails(), instanceOf(RefundAvailabilityUpdatedEventDetails.class));
+    }
```
Comment (jankowiakmaria) [inline src/test/java/uk/gov/pay/connector/events/model/refund/EventFactoryTest.java:301]: Inconsistent test naming (`should...for`, `given...should`).

**Review by jankowiakmaria — COMMENTED** [review]
One question, one minor comment (that doesn't need to be addressed as part of this PR).

Other than that LGTM, although I'm not sure about merging factories into one bigger class (good for sharing `createRefundAvailabilityUpdatedEvent`, but slightly harder to read - would be good to improve in the next iteration).

File: src/main/java/uk/gov/pay/connector/events/model/EventFactory.java (hunk 1 above)
Comment (mrlumbu) [reply src/main/java/uk/gov/pay/connector/events/model/EventFactory.java]: yeah this is wrong. thanks

**Review by jankowiakmaria — APPROVED** [review]
